instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/utils.py

```diff
@@ -64,6 +64,12 @@ def set_attributes_from_context(span, context):
         if key == "timelimit" and value in [(None, None), [None, None]]:
             continue
 
+        if key == "timelimit":
+            if value in [(None, None), [None, None]]:
+                continue
```

Comment:
Looks like this is already covered in above lines 64-65

Reply:
Thank you @lonewolf3739 for your comment, The case I want to handle here is when we set only one of the soft/hard limits in the celery task so we will get here `[None, Value]` or `[Value, None]` and this case isn't covered.

Reply:
Correct, I see that below. What I am saying is `if value in [(None, None), [None, None]]: continue` is a redundant check.

Reply:
Not sure I understand why, we can have three cases:
1. soft and hard limit was set - we will instrument them both.
2. Only one of the limits was set and we will instrument that one and the other will set to empty string.
3. No time limit was set - then we don't want to instrument them at all.

If I'll remove this check we will instrument the time limit with empty string even if they weren't set.

Reply:
I am saying that case is already covered [above](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/847/files#diff-672f16686553f36e3138aeaf3d146c467f3b6792448e59f393ac87c1f676bf23L62-L65)
```py
        # Skip `timelimit` if it is not set (it's default/unset value is a
        # tuple or a list of `None` values
         if key == "timelimit" and value in [(None, None), [None, None]]:
             continue
```

Reply:
Oh sorry @lonewolf3739 I missed these lines, I wanted to merge it to on `if` and forgot to remove the original ones.
I fixed that here 37e0a296cf33beb228f42ef600e832c36e907fcc.